docs/pages/push-notifications/receiving-notifications.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Receiving Notifications
+---
+
+You can now successfully send a notification to your app! If all you wanted was purely informational notifications, then you can stop here. But Expo provides the capabilities to do so much more: maybe you want to update the UI based on the notification, or maybe navigate to a particular screen if a notification was selected.
+
+Like most things with Expo, handling notifications is simple and straightforward across all platforms. All you need to do is add a listener using the [`Notifications` API](/versions/latest/sdk/notifications).
+
+```javascript
+import React from 'react';
+import { Text, View } from 'react-native';
+import * as Notifications from 'expo-notifications';
+
+// This refers to the function defined earlier in this guide, in Push Notifications Set Up
+import registerForPushNotificationsAsync from './registerForPushNotificationsAsync';
+
+/* @info This handler determines how your app handles notifications that come in while the app is foregrounded */
+Notifications.setNotificationHandler({
+  handleNotification: async () => ({
+    shouldShowAlert: true,
+    shouldPlaySound: false,
+    shouldSetBadge: false,
+  }),
+});
+/* @end */
+
+export default class App extends React.Component {
+  state = {
+    notification: {},
+    response: {},
+  };
+
+  componentDidMount() {
+    registerForPushNotificationsAsync();
+
+    /* @info This listener is fired whenever a notification is received while the app is foregrounded. */
+    this._notificationListener = Notifications.addNotificationReceivedListener(
+      this._handleNotification
+    );
+    /* @end */
+
+    /* @info This listener is fired whenever a user taps on or interacts with a notification (works when app is foregrounded, backgrounded, or killed). This listener is especially useful for routing users to a particular screen after they tap on a particular notification. */
+    this._notificationResponseListener = Notifications.addNotificationResponseReceivedListener(
+      this._handleNotificationResponse
+    );
+    /* @end */
+  }
+
+  _handleNotification = notification => {
+    /* @info Want to know what the format of the object this listener receives is? Find the specifics in the expo-notifications documentation */
+    this.setState({ notification: notification });
+    /* @end */
+  };
+
+  _handleNotificationResponse = response => {
+    /* @info Want to know what the format of the object this listener receives is? Find the specifics in the expo-notifications documentation */
+    this.setState({ response: response });
+    /* @end */
+  };
+
+  render() {
+    return (
+      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+        <Text>Origin: {this.state.notification.origin}</Text>
```

Comment:
This doesn't seem correct, according to the types and your code on: `docs/pages/push-notifications/overview.md` origin doesn't exist anymore and this code doesn't reflect the modifications of the notification object.
It should be or am i missing something?:
```
<View style={{ alignItems: 'center', justifyContent: 'center' }}>
      <Text>Title: {this.state.notification.request.content.title}</Text>
      <Text>Body: {this.state.notification.request.content.body}</Text>
      <Text>Data: {JSON.stringify(this.state.notification.request.content.data)}</Text>
</View>
```

Reply:
good catch! i forgot to update the UI for this example